src/Microsoft.AspNet.Mvc.Core/ActionDescriptorProviderContext.cs

```diff
@@ -3,15 +3,10 @@
 
 using System.Collections.Generic;
 
-namespace Microsoft.AspNet.Mvc
+namespace Microsoft.AspNet.Mvc.Core
```

Comment:
I went this way, but eventually decided that namespace changes need to happen all over the place, this suggestion makes sense, but since this PR is already very large I didn't want to make it bigger than it needs to be.

I used .Core for things in the root folder of MVC that shouldn't be user visible by default. (Like I mentioned above more are needed though)